### PR TITLE
[SYCL][E2E] Add virtual memory operations test for `sycl_ext_oneapi_virtual_mem` extension

### PR DESCRIPTION
**Pull request by DYNIO-INTEL**
Based on the test plan https://github.com/intel/llvm/pull/15509, this PR adds an e2e test that performs following checks:
- A check should be performed that we can successfully perform and immediately release a valid reservation of virtual memory.
- A check should be performed that methods `get_context()`, `get_device()` and `size()` return correct values (i.e. ones which were passed to physical_mem constructor).
- A check should be performed that a value returned from a valid call to `map()` is the same as `reinterpret_cast<void *>(ptr)`.
- A check should be performed that we can change access mode of a virtual memory range and immediately see it changed.
- A check should be performed that we can successfully map and immediately unmap a virtual memory range.

### CODE REVIEW

File: sycl/test-e2e/VirtualMem/virtual_mem_operations.cpp [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+    size_t BytesRequired = RequiredNumElements * sizeof(int);
+    size_t UsedGranularity = GetLCMGranularity(Device, Context);
+    size_t AlignedByteSize = GetAlignedByteSize(BytesRequired, UsedGranularity);
+    uintptr_t VirtualMemoryPtr =
+        syclext::reserve_virtual_mem(0, AlignedByteSize, Context);
+    syclext::physical_mem PhysicalMem{Device, Context, AlignedByteSize};
+    void *MappedPtr = PhysicalMem.map(VirtualMemoryPtr, AlignedByteSize,
+                                      syclext::address_access_mode::read_write);
+
+    syclext::set_access_mode(MappedPtr, AlignedByteSize,
```
Comment (maarquitos14) [inline sycl/test-e2e/VirtualMem/virtual_mem_operations.cpp:107]: Can we also check that before this change the access mode is `read_write` as set in `map`? Just to make sure that it wasn't `read` all the time.

**Review by maarquitos14 — APPROVED** [review]
Just a nit.

File: sycl/test-e2e/VirtualMem/virtual_mem_operations.cpp (hunk 1 above)
Comment (DYNIO-INTEL) [reply sycl/test-e2e/VirtualMem/virtual_mem_operations.cpp:107]: Sure, will be done.

**Comment by DYNIO-INTEL** [comment]
@intel/llvm-gatekeepers Can we merge it?

**Comment by sarnex** [comment]
we should wait until CI finishes

**Pull request merged**